api/service/version_service.go

```diff
@@ -54,20 +71,72 @@ func (service *versionsService) query() *gorm.DB {
 		Select("versions.*")
 }
 
-func (service *versionsService) ListVersions(ctx context.Context, modelID models.ID, monitoringConfig config.MonitoringConfig) (versions []*models.Version, err error) {
-	err = service.query().
-		Where(models.Version{ModelID: modelID}).
-		Order("created_at DESC").
-		Find(&versions).
-		Error
+func decodeCursor(cursor string) (cursorPagination, bool) {
```

Comment:
It will be better to move this method to its own file/package, it will be reused for other resources pagination too.

Reply:
Can we make it generalized though? somehow now it dependens to `versionID` and `versionModelID`

Reply:
sure, I'll try to generalized this approach